Plots/Polygons/NCL_polyg_19.py

```diff
@@ -0,0 +1,200 @@
+"""
+NCL_polyg_19.py
+===============
+This script illustrates the following concepts:
+   - Adding lines and polygons to a map
+   - Adding a map to another map as an annotation
+   - Coloring shapefile outlines based on an array of values
+   - Drawing a custom labelbar on a map
+   - Using functions for cleaner code
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/polyg_19.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/polyg_19_lg.png
+"""
+
+###############################################################################
+# Import packages:
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+from matplotlib.collections import PatchCollection
+from matplotlib.patches import Polygon
+import matplotlib.colors as colors
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+import matplotlib.cm as cm
+import shapefile as shp
+import numpy as np
+
+import geocat.datafiles as gdf
+from geocat.viz import util as gvutil
+
+###############################################################################
+# Read in data:
+
+file1 = open(gdf.get("shape_files/tl_2017_us_state.dbf"), 'r')
+file2 = open(gdf.get("shape_files/tl_2017_us_state.shp"), 'r')
+file3 = open(gdf.get("shape_files/tl_2017_us_state.shx"), 'r')
+file4 = open(gdf.get("shape_files/tl_2017_us_state.prj"), 'r')
+
+# Open the shapefile
+us = shp.Reader(gdf.get("shape_files/tl_2017_us_state.dbf"))
+
+# Open the text file with the population data
+statepopulationfile = open(gdf.get("ascii_files/us_state_population.txt"), 'r')
```

Comment:
I would refactor this and other variables in the file to include underscores between the words i.e. `state_population_file` or even shorten it to `state_pop_file`. This would make your code more readable.